### PR TITLE
Improved Deploy script

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Remove setting of properties that are unused by OpenShift infrastructure.
Remove unused Che properties and `DockerConnectorProvider`.

Rework `sed` commands not to expect default values from yaml file but setting always defaults that are needed for the corresponding flavour.
```
-    if [ "${ENABLE_SSL}" == "false" ]; then sed "s/    CHE_INFRA_OPENSHIFT_TLS__ENABLED: \"true\"/    CHE_INFRA_OPENSHIFT_TLS__ENABLED: \"false\"/" ; else cat -; fi | \
+    sed "s/    CHE_INFRA_OPENSHIFT_TLS__ENABLED:.*/    CHE_INFRA_OPENSHIFT_TLS__ENABLED: \"${ENABLE_SSL}\"/" | \
```
### What issues does this PR fix or reference?
-

#### Release Notes
N/A

### CODE REVIEW

**Comment by sleshchenko** [comment]
I don't see any usage of `remote-debugging-enabled` and `keycloak-disabled` values from config map. @davidfestal @ibuziuk Can you confirm that I can remove them in che6 branch?

File: dockerfiles/init/modules/openshift/files/scripts/che-openshift.yml [hunk 1]
```diff
@@ -95,35 +95,19 @@ items:
...
     che.predefined.stacks.reload_on_start: "true"
     log-level: INFO
-    docker-connector: openshift
     port: "8080"
     remote-debugging-enabled: "false"
     che-oauth-github-forceactivation: "true"
-    workspaces-memory-limit: 2300Mi
-    workspaces-memory-request: 1500Mi
     enable-workspaces-autostart: "false"
     keycloak-oso-endpoint: ${KEYCLOAK_OSO_ENDPOINT}
```
Comment (garagatyi) [inline dockerfiles/init/modules/openshift/files/scripts/che-openshift.yml]: Would be awesome to have all these values in the same format, e.g. KEYCLOAK_GITHUB_ENDPOINT

**Comment by sleshchenko** [comment]
@davidfestal @ibuziuk Also Can we remove annotations and labels that are not used in this yaml like https://github.com/eclipse/che/blob/9e7386e726011bf52b64389c492d274bd68fea5f/dockerfiles/init/modules/openshift/files/scripts/che-openshift.yml#L18-L25 https://github.com/eclipse/che/blob/9e7386e726011bf52b64389c492d274bd68fea5f/dockerfiles/init/modules/openshift/files/scripts/che-openshift.yml#L28-L31

File: dockerfiles/init/modules/openshift/files/scripts/che-openshift.yml (hunk 1 above)
Comment (sleshchenko) [reply dockerfiles/init/modules/openshift/files/scripts/che-openshift.yml]: I'll rework it

**Comment by davidfestal** [comment]
> I don't see any usage of `remote-debugging-enabled` and `keycloak-disabled` values from config map. @davidfestal @ibuziuk Can you confirm that I can remove them in che6 branch?

I can confirm that `keycloak-disabled` is not used anymore. Comes from the initial `rh-che` implementation.

On the contrary for the `remote-debugging-enabled`, it was used in the initial `deploy_che.sh` and yaml deployment descriptor to set the `CHE_DEBUG_SERVER` (see https://github.com/ibuziuk/docs/blob/master/che_remote_debugging.adoc) So I assume this could still be useful also now.

**Comment by davidfestal** [comment]
> @davidfestal @ibuziuk Also Can we remove annotations and labels that are not used in this yaml like

I assume you could remove quite a bunch of them, since the yaml from which this file was initially copied was a concatenated yaml generated by fabric8 for the fabric8 context. But I think that @l0rd and @ibuziuk would know better than me what can be removed.

**Comment by sleshchenko** [comment]
@davidfestal Thanks for your response.
As far as I understand Ilya used this `CHE_DEBUG_SERVER` env var for putting it as env var of Che CLI. But since deploy script doesn't use CLI for the launching of Che it can be removed.

**Comment by sleshchenko** [comment]
@davidfestal @ibuziuk @l0rd I've removed all annotations and labels except `app: che`. Notify me if they should be reverted.

**Review by benoitf — APPROVED** [review]

**Review by garagatyi — APPROVED** [review]
Looks good! Fewer lines better!

**Review by riuvshin — APPROVED** [review]

**Comment by sleshchenko** [comment]
I've move configuration properties from `CHE_SERVER_CONFIGURATION` env var to config map.
@akorneta Also removed the corresponding configuration properties from che-server configuration.
@riuvshin @garagatyi Please take a look one more time

**Review by garagatyi — APPROVED** [review]

**Review by riuvshin — APPROVED** [review]

**Review by l0rd — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Comment by sleshchenko** [comment]
I was wrong about CHE_DEBUG_SERVER because it is used in the `entrypoint.sh` of `Che` container https://github.com/eclipse/che/blob/980a378c1482ad60ed322046251df8663a587574/dockerfiles/che/entrypoint.sh#L157
@akorneta Will fix the launching of Che server in debug mode deployed by deploy script

**Pull request merged**